signer/spiffe/spiffe.go

```diff
@@ -45,3 +54,96 @@ func Signer(ctx context.Context, socketPath string) (cryptoutil.Signer, error) {
 
 	return cryptoutil.NewSigner(svid.PrivateKey, cryptoutil.SignWithIntermediates(svid.Certificates[1:]), cryptoutil.SignWithCertificate(svid.Certificates[0]))
 }
+
+func DelgatedSigner(ctx context.Context, socketPath string, workloadPath string) (cryptoutil.Signer, error) {
+	client, err := getDelgatedClient(ctx, socketPath)
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Printf("#####: %s", socketPath)
+
+	binPath, err := exec.LookPath(workloadPath)
+	if err != nil {
+		return nil, err
+	}
+
+	hasher := sha256.New()
+	s, err := ioutil.ReadFile(binPath)
```

Comment:
prefer os.ReadFile, ioutil is deprecated

Reply:
In actuality we should probably be using os.Open and io.Copy instead of reading the entire file into memory.